src/coreclr/nativeaot/BuildIntegration/Microsoft.DotNet.ILCompiler.SingleEntry.targets

```diff
@@ -33,22 +31,22 @@
     <IlcDynamicBuildPropertyDependencies>SetupProperties</IlcDynamicBuildPropertyDependencies>
   </PropertyGroup>
 
-  <!-- If called via package instead of the SDK, update the runtime package version to match the build package -->
-  <ItemGroup Condition="'$(AotRuntimePackageLoadedViaSDK)' != 'true'">
+  <!-- Update the KnownILCompilerPack version to match the PackageReference, in case there's an explicit reference to a specific version. -->
```

Comment:
If we drop this, https://github.com/dotnet/runtime/blob/main/docs/workflow/building/coreclr/nativeaot.md#building-packages and https://github.com/dotnet/runtime/blob/main/src/coreclr/nativeaot/docs/compiling.md#using-daily-should be updated with alternative instructions.

Reply:
One thing I'm worried about with this configuration is that we're effectively splitting the framework with this config, namely the ref assemblies are from one version and the runtime assemblies are from another.

Reply:
> One thing I'm worried about with this configuration is that we're effectively splitting the framework with this config,

Yes, it has potential versioning issues. These versioning issues are rarely problem in practice. The only 100% reliable way to avoid these potential versioning issues is by waiting for fully coherent SDK build. that happens like once a month.

Reply:
>  only 100% reliable way to avoid these potential versioning issues is by waiting for fully coherent SDK build. that happens like once a month.

I thought this was all done by targeting packs. Can we not engineer the SDK to pull the appropriate targeting pack based on a single property value? That is, you specify one version and it pulls a ref pack (if not locally available) from NuGet with that version, and then an ILCompiler pack the same way?

I suppose the alternative is to rely more on the VMR to accelerate the production of coherent SDK builds, which honestly doesn't sound too bad either.